sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -188,6 +210,28 @@ class GpuSortAggregateMeta(
             s"'complete', or 'all'")
       }
     }
+    hashAggReplaceMode match {
+      case "partial" => {
+        // In partial mode, if there are non distinct functions and multiple distinct functions
+        // non distinct functions are computed using First operator. The final result would be
+        // incorrect for non distinct functions for partition size > 1. Falling back to CPU for
+        // this special case.
+        if (agg.aggregateExpressions.exists(e => e.aggregateFunction.isInstanceOf[First])) {
+          val count = agg.aggregateExpressions.flatMap(expr =>
+            expr.children.flatMap {
+              _.collect {
+                case a: Count => a
+              }
+            })
+          if (count.size > 1) {
+            willNotWorkOnGpu("Aggregate of non distinct functions with multiple distinct " +
+              "functions is non deterministic for non distinct functions as it is computed using " +
+              "First.")
```

Comment:
Wondering if there might be a place in aggregateExpressions to consolidate this fall back?

Reply:
There is duplicate code in other sections as well in  GpuSortAggregateMeta and GpuHashAggregateMeta. Can we move common things to some Util method if it makes sense?

Reply:
Sure that is ok for now. Maybe a follow on. I would however, check for sort aggregate we actually do fall back.